### ISSUE TITLE
android calling xdrip with miaomiao packets after very long time.

### ISSUE THREAD

**Issue by tzachi-dar** [NightscoutFoundation/xDrip#730]
Here are the pictures that have been reported on facebook:

![image](https://user-images.githubusercontent.com/7942372/50564723-2a40ec80-0d30-11e9-90ae-87ff2c080808.png)

**Comment by tzachi-dar** [comment]
This strange data pattern was reported by the person more than once.

Looking at his xdrip db the following pattern was observed: 

select datetime(timestamp / 1000, 'unixepoch', 'localtime'), timestamp, calculatedbg from LibreBlock order by timestamp

"2018-12-27 21:38:14"	"1545939494253"	"137.155660799969"
"2018-12-27 21:43:14"	"1545939794067"	"130.833726915018"
"2018-12-28 04:58:35"	"1545965915093"	"130.833726915018"
"2018-12-28 04:58:57"	"1545965937272"	"0.0"
"2018-12-28 05:06:34"	"1545966394950"	"176.646097122402"
"2018-12-28 05:11:34"	"1545966694755"	"173.008820092704"

So, we can see that there are no readings from the 27/12 - 21:43 until 28-12 04:58

That said the readings when they start connecting seems strange. one of them is 130 and the other is already 176.

Looking into the blocks it seems that they come from two different times from the sensor life.
In other words, there is the following two readings that are consistent:

       MS                        time                        internal counter
1545939794067 2018_12_27_21_43_14           8013 glucoseLevel =  157.6 
1545965937272 2018_12_28_04_58_57           8446 glucoseLevel =  204.3 

The two readings above seems right since they have about 7 hours of difference between them, but on the other hand the libre counter is 427 minutes different (8446-8013)

While we also have this reading.
1545965915093 2018_12_28_04_58_35          8018                  glucoseLevel =  157.6 
As we can see on the last reading time is like the later reading but counter is from 7 hours ago.
This causes our code to think that this is a recent packet.
More than that, since no data existed for many hours, we had two parts in the code trying to complete the gap, and this is how the problem has happened.

**Comment by tzachi-dar** [comment]
How to fix?

Since we set the time very early in the code that handles the packet, and also start by taking a wakelock on this datapath, I'm not sure there is much we can do.

One thing that might help is if on the function 
public void onCharacteristicChanged(BluetoothGatt gatt, final BluetoothGattCharacteristic characteristic)
we can get the time of the data. I did not find such a place.

The other option is to store on the sensor the exact start time (based on libre data) and throw away packets that don't match the expected time.
